src/services/project.service.js

```diff
@@ -46,13 +49,40 @@ const updateProjectById = async (id, teacher, body) => {
 
   Object.assign(project, { state: body.state });
   await project.save();
-
   return project;
 };
 
+const getAllMessages = async (id, teacher) => {
+  const project = await Project.findOne({ _id: id }).populate('classes messages');
+
+  const teacherIds = project && project.classes.map((c) => String(c.teacher));
+  if (project && teacherIds.indexOf(teacher) < 0) {
+    throw new ApiError(httpStatus.FORBIDDEN, 'No class belongs to the teacher');
+  }
+  return project.messages;
+};
+
+const addMessage = async (id, messageBody, teacher) => {
+  const project = await getProjectById(id, teacher);
+  const teacherIds = project && project.classes.map((c) => String(c.teacher));
+  if (teacherIds.indexOf(teacher) < 0 || teacherIds.indexOf(messageBody.to) < 0) {
+    throw new ApiError(httpStatus.FORBIDDEN, 'message invalid');
+  }
+  const message = messageService.createMessage(messageBody);
+  await Project.updateOne({ _id: id }, { $push: { messages: message._id } });
+  const senderNotification = sendMessage(await userService.getUserById(messageBody.to), messageBody.message);
+  const receiverNotification = receiveMessage(await userService.getUserById(messageBody.from), messageBody.message);
+  await notificationService.sendNotification(messageBody.from, senderNotification);
+  await notificationService.sendNotification(messageBody.to, receiverNotification);
```

Comment:
i dont think the sender needs to receive a notification?

Reply:
ok, then remove it